packages/type-utils/src/containsTypesByName.ts

```diff
@@ -5,13 +5,16 @@ import { isTypeFlagSet } from './typeFlagUtils';
 
 /**
  * @param type Type being checked by name.
+ * @param allowAny Whether to consider `any` and `unknown` to match.
  * @param allowedNames Symbol names checking on the type.
- * @returns Whether the type is, extends, or contains all of the allowed names.
+ * @param mustMatchAll Whether all parts have to match, as opposed to any parts matching.
+ * @returns Whether the type is, extends, or contains the allowed names (or all matches the allowed names, if mustMatchAll is true).
  */
-export function containsAllTypesByName(
+export function containsTypesByName(
```

Comment:
`@typescript-eslint/type-utils` is a public package so:
- this rename is a breaking change
- adding a new non-optional param is a breaking change

we'll need to:
- either keep the function named as is, or add an alias to maintain backwards compat
- set a default value for the new param (probably `true`)

Reply:
What if I add `containsAllTypesByName` back, making it call the new, renamed version passing in `true` to the 4th param. That will maintain the same behavior and keep it backwards compatible.

Reply:
I switched the behavior around so the fourth param is now `matchAnyInstead`, with default false to maintain the same behavior.